doc/arch/015-online-patient-lookup.md

```diff
@@ -0,0 +1,121 @@
+# Online Patient Lookup
+
+## Status
+
+Proposed on 2020-04-12
+
+## Context
+
+Want want to ensure that a patient’s medical records within the state
+are always available to the nurse for treatment if they are connected
+to the internet. See related
+[PRD](https://docs.google.com/document/d/1q6cppByQULfh3_mMXC4BJpiNN9Uc_awA6rreeEtUBaM/edit#)
+for more details on the feature and related specifications.
+
+The lookup will be a new API on the server that the mobile app will
+call. The following aspects of the problem are addressed in this ADR:
+
+- API contract
+- Data retention
+- Access restrictions
+- Audit logging
+- Rate limiting
+
+## Decision
+### Data retention
+There are two broad kinds of data retention on the app at a facility:
+1. Permanent: patient is within sync criteria: the same block, has an
+   appointment, or is assigned to the facility
+2. Temporary: patient is not within the sync criteria
+
+In order to have a better control of the retention, we will send this
+information from the server. This will be a static number to begin
+with, but can later vary depending on the state, country or other
+factors.
+
+We will implement temporary retention in the app with this
+feature. With every patient retrieved via the lookup API, we will:
+
+- ensure the data is synced to the server
+- delete the data after the retention time period has passed
+
+< @mobile-devs: please add details on possible approaches here >
+
+Note that the retention time period is not strict. i.e,
+it is okay for the record to stay on the phone for 2h if the retention
+period is 1h. We can perform deletions at a convenient time as long as
+the deletion happens deterministically.
+
+_Alternatively_: we can choose to hard-code the retention period on
+the app.
+
+### API contract
+For the request, we will use the endpoint `GET`:
+https://api.simple.org/api/v4/patient/identifier/, where `identifier`
```

Comment:
`patients`, since we're returning a list? I'm not sure we can nail down a strictly "RESTful" URL for this endpoint 😅 

The endpoint itself is a simple change to make while developing, so finding the right URL isn't a blocker for the ADR, but it is a blocker for shipping the endpoint 👍

Reply:
Yeah, `patients` makes sense. I specifically put the endpoint and API design here because I wanted feedback on it 😁 . Thanks!

Reply:
Changed this accordingly.